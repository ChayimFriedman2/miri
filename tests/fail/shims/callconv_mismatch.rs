extern "Rust" {
    fn pipe(fds: *mut std::ffi::c_int) -> std::ffi::c_int;
}

// Test the error for calling convention mismatch. 
fn main() {
    let mut fds = [-1, -1];
    let res = unsafe { pipe(fds.as_mut_ptr()) };
    //~^ ERROR: calling a function with calling convention C using caller calling convention Rust
}