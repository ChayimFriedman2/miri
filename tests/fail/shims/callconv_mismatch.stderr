error: Undefined Behavior: calling a function with calling convention C using caller calling convention Rust
  --> tests/fail/shims/callconv_mismatch.rs:LL:CC
   |
LL |     let res = unsafe { pipe(fds.as_mut_ptr()) };
   |                        ^^^^^^^^^^^^^^^^^^^^^^ calling a function with calling convention C using caller calling convention Rust
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE:
   = note: inside `main` at tests/fail/shims/callconv_mismatch.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error

